### PR TITLE
[www] Update links to use`localizedLink`

### PR DESCRIPTION
**Pull request by samovertonjr**
@tesseralis I'm not to familiar with how to go about linting. Is there something I can reference in Gatsby? Also, should we change download links that are in the brand guidelines to use localizedLink? 

## Description

 Updated all links in www to use `localizedLink` component.

## Related Issues

Addresses #20797

### CODE REVIEW

**Comment by tesseralis** [comment]
@samovertonjr Which download links are you talking about specifically?

As for linting, this is a good place to start: https://eslint.org/docs/rules/no-restricted-imports

**Comment by samovertonjr** [comment]
@tesseralis The specific file is in this [zip](https://www.gatsbyjs.org/Gatsby-Logos.zip).

**Comment by samovertonjr** [comment]
@tesseralis I added the new linting rule. I had to update the eslintignore file because all of www was being ignore.

File: www/.eslintrc.json [hunk 1]
```diff
@@ -1,5 +1,19 @@
 {
   "globals": {
-    "graphql": true
+    "graphql": false
```
Comment (tesseralis) [inline www/.eslintrc.json]: why was this changed?

File: www/gatsby-config.js [hunk 2]
```diff
@@ -55,15 +55,17 @@ if (process.env.AIRTABLE_API_KEY) {
...
-    ...langs.map(({ code }) => ({
-      resolve: `gatsby-source-git`,
-      options: {
-        name: `docs-${code}`,
-        remote: `https://github.com/gatsbyjs/gatsby-${code}.git`,
-        branch: `master`,
-        patterns: `docs/tutorial/**`,
-      },
-    }))
+    ...langs.map(({ code }) => {
```
Comment (tesseralis) [inline www/gatsby-config.js]: ...is this necessary?

**Comment by tesseralis** [comment]
One thing to do if you haven't: run `yarn lint` and make sure that the new rule works. i.e. change one of the localized links back to "Link" and make sure that the linting fails.

File: www/.eslintrc.json (hunk 1 above)
Comment (samovertonjr) [reply www/.eslintrc.json]: this was a mistake changing it back now.

File: www/gatsby-config.js (hunk 2 above)
Comment (samovertonjr) [reply www/gatsby-config.js]: This was came from eslint as a linting error. I just changed it to fix it.

**Comment by samovertonjr** [comment]
I ran yarn lint checking `import { Link } from 'gatsby'` and I got the error so it is working correctly. Thanks for all of your help

**Comment by samovertonjr** [comment]
@tesseralis Should I be pointing this branch to a different base branch?

**Comment by tesseralis** [comment]
@samovertonjr master branch should be fine!

**Comment by samovertonjr** [comment]
@tesseralis I don't why my test are failing for CircleCI and locally I don't get any errors

**Comment by LekoArts** [comment]
Got some more merge conflicts

**Comment by marcysutton** [comment]
Are there tests for localizedLink, @tesseralis ? This is a really broad change, so I want to make sure we’ve got test coverage for it.

**Comment by tesseralis** [comment]
@marcysutton that's a good point. There aren't right now, but I can make some and merge those in before this PR gets merged.

**Comment by tesseralis** [comment]
@samovertonjr, could you update this PR by fixing the merge conflicts? Once that's done we can merge this in :)

**Comment by tesseralis** [comment]
The linting is failing right now. Could you make sure to run `yarn format` or run pre-commit hooks?

**Comment by tesseralis** [comment]
So it seems like the linting is picking up stuff that it should have picked up before and changing files that aren't related to this PR. For now, I think we should:

1. remove the changes to the linters (including adding the rule to not import gatsby link) and revert all the files that aren't related to changing the `Link` component.
2. Open up a new PR to fix the linter issue. If you'd like to do this, go ahead, otherwise someone from the Gatsby core team can do it :)

**Comment by samovertonjr** [comment]
@tesseralis I will do both. Sorry I haven't been as responsive but I have time later today to fix this issue. Thanks for your patience with me 😄.

**Comment by tesseralis** [comment]
It's okay! Thank you for all your hard work and my apologies for adding on all this extra work :)
